modules/service-account-auto-secret-removed.adoc

```diff
@@ -0,0 +1,23 @@
+// Module included in the following assemblies:
+//
+// * authentication/using-service-accounts-in-applications.adoc
+// * pods/nodes-pods-secrets.adoc
+
+:_content-type: CONCEPT
+[id="auto-generated-sa-token-secrets_{context}"]
+= About automatically generated service account token secrets
+
+When a service account is created, a service account token secret is automatically generated for it. This service account token secret, along with an automatically generated Docker configuration secret, is used to authenticate to the internal {product-title} registry. Do not rely on these automatically generated secrets for your own use; they might be removed in a future {product-title} release.
+
+[NOTE]
+====
+Prior to {product-title} 4.11, a second service account token secret was generated when a service account was created. This service account token secret was used to access the Kubernetes API.
+
+Starting with {product-title} 4.11, this second service account token secret is no longer created. This is because the `LegacyServiceAccountTokenNoAutoGeneration` upstream Kubernetes feature gate was enabled, which stops the automatic generation of secret-based service account tokens to access the Kubernetes API.
+
+After upgrading to {product-version}, any existing service account token secrets are not deleted and continue to function.
+====
+
+If your workload needs a service account token, use the `TokenRequest` API to request a bound service account token. Tokens obtained from the `TokenRequest` API are more secure than service account token secrets, because they have a bounded lifetime and are not readable by other API clients. For more information, see _Creating a service account token secret_.
```

Comment:
> use the `TokenRequest` API to request a bound service account token

Same as comment https://github.com/openshift/openshift-docs/pull/62197#discussion_r1267711189

> are not readable by other API clients

It seems confusing and meaningless, maybe we should delete it.

Reply:
@HanielZx if we take out that part, then that would only leave having a bounded lifetime as the reason why these tokens are more secure than service account token secrets. I had pulled this wording from elsewhere in the docs. See the note here: https://docs.openshift.com/container-platform/4.13/nodes/pods/nodes-pods-secrets.html#nodes-pods-secrets-creating-sa_nodes-pods-secrets

If you don't want to say that, can you recommend what else we should say instead, on why this is the preferred method? Or how we could change this to be clearer to what we mean?

Reply:
> because they have a bounded lifetime and are not readable by other API clients.

@bergerhoffer  Hi, I saw there were three pros provided by devs: https://github.com/openshift/openshift-docs/pull/61541#discussion_r1250767436
The first pro "expire with time" corresponds to "they have a bounded lifetime". I think maybe we can add other two pros to make it more clearer.

Reply:
@HanielZx no, that is purposefully pointing to "Creating a service account token secret".

At first, we talk about requesting bound SA tokens. And for more information for that, we say to go see "Using bound service account tokens", which contains the procedure on creating bound SA tokens using volume projection.

But then after that, we say:

> You can also manually create a service account token secret to create a token, if the security exposure of a non-expiring token in a readable API object is acceptable to you. For more information, see _Creating a service account token secret_.

This one is talking about that if you want to manually create a SA token secret (that doesn't have an expiration, etc.), then go see the instructions in "Creating a service account token secret".

Let me know if that makes sense, or if you have any further questions on it

Reply:
@bergerhoffer Get it, LGTM.